### ISSUE TITLE
Configurable language server close timeout

### ISSUE THREAD

**Issue by snosratiershad** [helix-editor/helix#11542]
I want to increase close language server timeout to prevent force shutdown shutdown with `Timed out waiting for language servers to shutdown` error message.
I found that there is a static 3000ms timeout in code (helix-view/src/editor.rs:2023). I think the best approach is to use specific timeout for each language-server but also a general configurable timeout is better.

**Comment by CietyGoblin** [comment]
brew install helix

**Comment by ArmanSoltani** [comment]
I can take this one if nobody is working on it.

**Comment by ArmanSoltani** [comment]
I agree with you that a per language-server `close_timeout` config makes more sense given that it should only be needed to increase the default one for very slow language-server. Also, it makes sense to me to put that config in the `[language-server.mylang]` group.

@snosratiershad just out of curiosity do you encounter that timeout for a specific language-server or for all of them in you machine?

**Comment by snosratiershad** [comment]
@ArmanSoltani for specific lsps

**Comment by the-mikedavis** [comment]
See https://github.com/helix-editor/helix/pull/5266
